src/psycop_feature_generation/loaders/raw/utils.py

```diff
@@ -131,6 +135,12 @@ def load_from_codes(
         + f" FROM [fct].{fct} WHERE {source_timestamp_col_name} IS NOT NULL AND ({match_col_sql_str})"
     )
 
+    if administration_method:
+        sql += f" AND type_kodetekst = '{administration_method}'"
+
+    if administration_route:
+        sql += f" AND admvej_kodetekst = '{administration_route}'"
```

Comment:
Looks great! Would be super helpful if we guard against invalid `administration_methods`, so users are warned if they set requirements that will return no rows.

```suggestion
    if administration_method:
        allowed_administration_methods = ("OR", "IM", "IV")
        if administration_method not in allowed_administration_methods:
            raise ValueError(f"Administration method is not allowed, allowed values are {allowed_administration_methods}")
        sql += f" AND type_kodetekst = '{administration_method}'"
```

Similar for `administration_route`.

Does that make sense? :-)

Reply:
Yes, agreed. great idea!